### PR TITLE
Support targeting multiple entities with UiEvents generated by the MouseUiSystem

### PR DESCRIPTION
**Pull request by bonsairobo**
## Description

Support targeting multiple entities with UiEvents generated by the MouseUiSystem

## Additions

- Add the `transparent_target: bool` field to `UiTransform` and `UiTransformData`.

## Modifications

- Use the field in `ui::events::targeted` to support the desired behavior.

## PR Checklist

By placing an x in the boxes I certify that I have:

- [ ] Updated the content of the book if this PR would make the book outdated.
- [x] Added a changelog entry if this will impact users, or modified more than 5 lines of Rust that wasn't a doc comment.
- [ ] Added unit tests for new code added in this PR.
- [x] Acknowledged that by making this pull request I release this code under an MIT/Apache 2.0 dual licensing scheme.

If this modified or created any rs files:

- [x] Ran `cargo +stable fmt --all`
- [x] Ran `cargo clippy --all --features "empty"`
- [x] Ran `cargo test --all --features "empty"`

### CODE REVIEW

File: amethyst_ui/src/event.rs [hunk 1]
```diff
@@ -153,63 +153,72 @@ impl<'a, T: BindingTypes> System<'a> for UiMouseSystem<T> {
...
+        .map(|(e, t, _m, _, _)| (e, t))
+        .collect();
+    entity_transforms.sort_by(|(_, t1), (_, t2)| {
+        t2.global_z
+            .partial_cmp(&t1.global_z)
+            .expect("Unexpected NaN")
+    });
+
+    let mut first_opaque = None;
+    for (i, (_e, t)) in entity_transforms.iter().enumerate() {
```
Comment (azriel91) [inline amethyst_ui/src/event.rs]: Perhaps this is clearer:

```rust
let first_opaque = entity_transforms.iter().position(|(_e, t)| t.opaque);
```

Comment (bonsairobo) [reply amethyst_ui/src/event.rs]: Definitely better, thanks!

**Comment by bonsairobo** [comment]
Oops. I think I made a git noob mistake. Not sure why I have duplicate commits.

**Comment by azriel91** [comment]
bors retry

**Pull request merged**